core/src/commonMain/kotlin/com/ustadmobile/core/domain/validateusername/ValidateUsernameUseCase.kt

```diff
@@ -0,0 +1,41 @@
+package com.ustadmobile.core.domain.validateusername
+
+/**
+ * Validates whether a username meets all required criteria:
+ * - Must not be too short or too long
+ * - Must not start with a number
+ * - Must only contain valid characters (letters, numbers, dots, underscores)
+ */
+class ValidateUsernameUseCase {
+
+    enum class ValidationResult {
+        VALID,
+        INVALID_TOO_SHORT,
```

Comment:
I think we need to show an error message specific to the validation result: e.g. if the username is rejected because it is too short, we should show only "Invalid: must be at least 3 characters". Many of our users are a bit unfamiliar with tech, so I think it would be better to tell them what is wrong with the username as they entered it, rather than make them read all the rules and figure it out. 

You can add a StringResource to the ValidationResult (e.g. ```ValidationResult(val errorMessage: StringResource?)```).

Then in viewModel, you can use it e.g.

```
val validationResult = validateUsernameUseCase(...)
if(validationrResult != Valid) {
    usernameError = validationResult.errorMessage?.let { systemImpl.get(it} }
}